client/marshaller/marshaller-tests/src/main/java/org/infinispan/marshaller/test/CompatibilityCacheFactory.java

```diff
@@ -0,0 +1,215 @@
+package org.infinispan.marshaller.test;
+
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killRemoteCacheManager;
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killServers;
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.startHotRodServer;
+import static org.infinispan.server.memcached.test.MemcachedTestingUtil.killMemcachedClient;
+import static org.infinispan.server.memcached.test.MemcachedTestingUtil.killMemcachedServer;
+import static org.infinispan.server.memcached.test.MemcachedTestingUtil.startMemcachedTextServer;
+import static org.infinispan.test.TestingUtil.killCacheManagers;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Collections;
+
+import org.apache.commons.httpclient.HttpClient;
+import org.infinispan.Cache;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.commons.api.BasicCacheContainer;
+import org.infinispan.commons.api.Lifecycle;
+import org.infinispan.commons.equivalence.Equivalence;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.rest.NettyRestServer;
+import org.infinispan.rest.configuration.RestServerConfigurationBuilder;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.memcached.MemcachedServer;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+
+import net.spy.memcached.ConnectionFactory;
+import net.spy.memcached.ConnectionFactoryBuilder;
+import net.spy.memcached.DefaultConnectionFactory;
+import net.spy.memcached.MemcachedClient;
+import net.spy.memcached.transcoders.Transcoder;
+
+/**
+ * Compatibility cache factory taking care of construction and destruction of
+ * caches, servers and clients for each of the endpoints being tested.
+ *
+ * @author Galder Zamarreño
+ * @author Ryan Emerson
+ */
+final public class CompatibilityCacheFactory<K, V> {
```

Comment:
This seems like a duplicate of the `compatibility` integration testsuite. Could you depend on `infinispan-compatibility-mode-it` and avoid dependency?